trl/trainer/sft_trainer.py

```diff
@@ -219,6 +219,17 @@ def __init__(
             preprocess_logits_for_metrics=preprocess_logits_for_metrics,
         )
 
+        if self.args.max_steps > 0 and packing:
+            warnings.warn(
+                "You passed `packing=True` to the SFTTrainer, and you are training your model with `max_steps` strategy. Will force-set `infinite` argument of the packed dataset to `True` "
+            )
+            self.train_dataset.infinite = True
+        elif self.args.max_steps == -1 and packing:
+            warnings.warn(
```

Comment:
I think in that case we don't have to warn - the model will train as expected, no?

Reply:
Makes sense, removed it